optuna/notebooks/jupytercon_demo/RAPIDS_xfeat_Optuna-GPU.ipynb

```diff
@@ -319,48 +313,67 @@
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
+   "execution_count": 8,
    "metadata": {},
    "outputs": [],
    "source": [
-    "def train_and_eval(df, penalty='l2', C=1.0, l1_ratio=None, fit_intercept=True, selector=None, return_model=False):\n",
-    "    # Splitting data and prepping for selector fit\n",
-    "    X_train,  X_test, y_train, y_test = train_test_split(df,\n",
-    "                                                         TARGET_COL,\n",
-    "                                                         random_state=np.random.seed(0),\n",
-    "                                                         shuffle=True)\n",
-    "    # Xfeat's internal fold mechanism creates RangeIndex references, so we need to do an index reset on our data frames.\n",
-    "    X_train = X_train.reset_index(drop=True)\n",
-    "    X_test = X_test.reset_index(drop=True)\n",
-    "    y_train = y_train.reset_index(drop=True)\n",
-    "    y_test = y_test.reset_index(drop=True)\n",
-    "    \n",
+    "def train_and_eval(df_train, df_test, penalty='l2', C=1.0, l1_ratio=None, fit_intercept=True, selector=None, return_model=False):\n",
+    "    \"\"\"\n",
+    "        Split the dataframe based on TARGET_COL\n",
+    "        Accepts the parameters to be set for the Logistic Regression model\n",
+    "        \n",
+    "        Evaluates on the split data and returns AUC score.\n",
+    "        \n",
+    "        Parameters\n",
+    "        ----------\n",
+    "        df_train, df_test: DataFrame to use for training and validation\n",
+    "        \n",
+    "        penalty, C, l1_ratio, fit_intercept: Parameters for the LogisticRegression Model\n",
+    "            For details refer to the documentation\n",
+    "\n",
+    "        selector: xfeat selector passed via Optuna (default=None)\n",
+    "            When set, the slector is used to fit and transform the data\n",
+    " \n",
+    "        return_model: Returns the fit model back to the calling fucntion (default=False)\n",
+    "        \n",
+    "        Returns\n",
+    "        ----------\n",
+    "        score: AUC score of the validation set\n",
+    "        classifier, signature: Returned only when return_model is set to True\n",
+    "    \"\"\"    \n",
     "    if selector:\n",
-    "        # For the selector, the label also needs to be in the DF\n",
-    "        X_train[TARGET_COL] = y_train\n",
-    "        X_test[TARGET_COL] = y_test\n",
+    "        # Getting the label column as it is dropped in the selector\n",
+    "        y_train = df_train[TARGET_COL]\n",
```

Comment:
kind of funny that in one branch you do y first and in the next branch do X first